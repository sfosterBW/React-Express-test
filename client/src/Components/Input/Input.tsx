import React, { FC } from 'react'
import styles from './Input.module.css'

interface InputTextProps {
  label: string
  name: string
  onChange: (event: any) => void
  value: string
}

export const InputText: FC<InputTextProps> = ({
  onChange,
  label,
  name,
  value
}) => {
  return (
    <div className={styles.wrapper} data-testid="wrapper">
      <label className={styles.label} htmlFor={name}>
        {label}
      </label>
      <input
        className={styles.text}
        data-testid="input-text"
        id={name}
        onChange={onChange}
        name={name}
        type="text"
        value={value}
      />
    </div>
  )
}

interface InputCheckboxProps {
  label: string
  name: string
  onChange: (event: any) => void
  value: boolean
}

export const InputCheckbox: FC<InputCheckboxProps> = ({
  onChange,
  label,
  name,
  value
}) => {
  const styleBackground = (): string | undefined => {
    const color: string = '#1ab545'
    return value ? color : undefined
  }

  return (
    <div className={styles.wrapper} data-testid="wrapper">
      <label className={styles.label} htmlFor={name}>
        {label}
      </label>
      <div className={styles.checkboxWrapper}>
        <input
          checked={value}
          className={styles.checkbox}
          data-testid="input"
          id={name}
          onChange={onChange}
          name={name}
          type="checkbox"
          value={String(value)}
        />
        <label
          style={{background: styleBackground()}}
          className={styles.checkboxLabel}
          data-testid="toggle"
          htmlFor={name}
        >
          <span className={styles.checkboxButton} />
        </label>
      </div>
    </div>
  )
}
