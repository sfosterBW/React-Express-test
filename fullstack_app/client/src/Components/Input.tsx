import React, { FC } from 'react'

interface InputTextProps {
  handleChange: (event: any) => void
  label: string
  name: string
  value: string
}

export const InputText: FC<InputTextProps> = ({
  handleChange,
  label,
  name,
  value }) => {

  return (
    <div className="input-wrapper">
      <label htmlFor={name}>{label}</label>
      <input
        onChange={() => handleChange}
        name={name}
        type="text"
        value={value} />
    </div>
  )
}

interface InputCheckboxProps {
  checked: boolean
  handleChange: (event: any) => void
  label: string
  name: string
  value: string
}

export const InputCheckbox: FC<InputCheckboxProps> = ({
  checked,
  handleChange,
  label,
  name,
  value }) => {

  return (
    <div className="input-wrapper">
      <label htmlFor={name}>{label}</label>
      <input
        checked={checked}
        onChange={() => handleChange}
        name={name}
        type="checkbox"
        value={value} />
    </div>
  )
}
