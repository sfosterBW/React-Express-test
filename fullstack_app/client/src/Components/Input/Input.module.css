.wrapper {
  display: flex;
  margin: 3vmin 0;
}

.label {
  display: inline-block;
  width: 16vmin;
  margin-right: 2vmin;
  font-weight: bold;
  text-align: right;
  font-size: calc(10px + 1vmin);
  align-self: center;
}

.text {
  height: calc(20px + 1vmin);
  max-width: 60vw;
  font-size: calc(12px + 1vmin);
  border: none;
  border-radius: 0.5vmin;
  padding: 0 1vmin;
}

.text:hover {
  box-shadow: 0.2vmin 0.2vmin 0.5vmin 0.5vmin rgba(0, 0, 0, 0.25);
}

.checkboxWrapper {
  display: flex;
  box-sizing: border-box;
}

.checkbox {
  height: 0;
  width: 0;
  visibility: hidden;
}

.checkboxLabel {
  display: flex;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  width: calc(30px + 3vmin);
  height: calc(16px + 2vmin);
  background: grey;
  border-radius: calc(4px + 2vmin);
  position: relative;
  transition: background-color .2s
}

.checkboxButton {
  content: '';
  position: absolute;
  top: calc(1px + 0.5vmin);
  left: calc(1px + 0.5vmin);
  width: calc(16px + 0.8vmin);
  height: calc(16px + 0.8vmin);
  border-radius: calc(10px + 2vmin);
  transition: 0.2s;
  background: #fff;
  box-shadow: 0 0 0.2vmin 0 rgba(10, 10, 10, 0.29)
}

.checkboxButton:hover {
  box-shadow: 0.2vmin 0.2vmin 1vmin 1vmin rgba(0, 0, 0, 0.25);
}

.checkbox:checked + .checkboxLabel .checkboxButton {
  left: calc(100% - 0.2vmin);
  transform: translateX(-100%);
}
