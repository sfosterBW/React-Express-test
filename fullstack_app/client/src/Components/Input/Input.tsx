import React, { FC } from 'react'
import styles from './Input.module.css'

interface InputTextProps {
  handleChange: (event: any) => void
  label: string
  name: string
  value: string
}

export const InputText: FC<InputTextProps> = ({
  handleChange,
  label,
  name,
  value
}) => {
  return (
    <div className={styles.wrapper} data-testid="wrapper">
      <label className={styles.label} htmlFor={name}>
        {label}
      </label>
      <input
        className={styles.text}
        data-testid="input-text"
        id={name}
        onChange={handleChange}
        name={name}
        type="text"
        value={value}
      />
    </div>
  )
}

interface InputCheckboxProps {
  checked: boolean
  handleChange: (event: any) => void
  label: string
  name: string
}

export const InputCheckbox: FC<InputCheckboxProps> = ({
  checked,
  handleChange,
  label,
  name
}) => {
  const styleBackground = (): string | undefined => {
    const color: string = '#1ab545'
    return checked ? color : undefined
  }

  return (
    <div className={styles.wrapper} data-testid="wrapper">
      <label className={styles.label} htmlFor={name}>
        {label}
      </label>
      <div className={styles.checkboxWrapper}>
        <input
          checked={checked}
          className={styles.checkbox}
          data-testid="input"
          id={name}
          onChange={handleChange}
          name={name}
          type="checkbox"
          value={String(checked)}
        />
        <label
          style={{background: styleBackground()}}
          className={styles.checkboxLabel}
          data-testid="toggle"
          htmlFor={name}
        >
          <span className={styles.checkboxButton} />
        </label>
      </div>
    </div>
  )
}
