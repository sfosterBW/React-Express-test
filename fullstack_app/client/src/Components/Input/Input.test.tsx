
import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import { InputText, InputCheckbox } from './Input'

describe('the InputText component', () => {
  const mockFunction = jest.fn()
  const label = "This is a label"
  const name = "This is a name"
  const value = "This is a value"
  const component = <InputText
    handleChange={mockFunction()}
    label={label}
    name={name}
    value={value} />
  const wrapper = mount(component)

  it('renders with the correct structure', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper.find('div')).toHaveLength(1)
    expect(wrapper.find('label')).toHaveLength(1)
    expect(wrapper.find('label').text()).toEqual(label)
    expect(wrapper.find('label').props().htmlFor).toEqual(name)
    expect(wrapper.find('input')).toHaveLength(1)
    expect(wrapper.find('input').props().name).toEqual(name)
    expect(wrapper.find('input').props().value).toEqual(value)
    expect(wrapper.find('input').props().type).toEqual("text")
  })

  it('functions as expected', () => {
    expect(wrapper.props().name).toEqual(name)
    expect(wrapper.props().label).toEqual(label)
    expect(wrapper.props().value).toEqual(value)
    wrapper.simulate('change')
    expect(mockFunction).toHaveBeenCalled()
  })

  it('renders the same as last time', () => {
    const tree = renderer
      .create(component)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})



describe('the InputCheckbox component', () => {
  const mockFunction = jest.fn()
  const checked = false
  const label = "This is a label"
  const name = "This is a name"
  const component = <InputCheckbox
    checked={checked}
    handleChange={mockFunction()}
    label={label}
    name={name} />
  const wrapper = mount(component)

  it('renders with the correct structure', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper.find('div')).toHaveLength(2)
    expect(wrapper.find('label')).toHaveLength(2)
    expect(wrapper.find('label').at(0).text()).toEqual(label)
    expect(wrapper.find('label').at(0).props().htmlFor).toEqual(name)
    expect(wrapper.find('input')).toHaveLength(1)
    expect(wrapper.find('input').props().checked).toEqual(checked)
    expect(wrapper.find('input').props().name).toEqual(name)
    expect(wrapper.find('input').props().type).toEqual("checkbox")
  })

  it('functions as expected', () => {
    expect(wrapper.props().label).toEqual(label)
    expect(wrapper.props().name).toEqual(name)
    expect(wrapper.props().checked).toEqual(checked)
    wrapper.simulate('change')
    expect(mockFunction).toHaveBeenCalled()
  })

  it('renders the same as last time', () => {
    const tree = renderer
      .create(component)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
