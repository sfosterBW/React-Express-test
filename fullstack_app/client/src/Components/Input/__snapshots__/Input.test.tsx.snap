// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the InputCheckbox component renders the same as last time 1`] = `
<div
  className="wrapper"
>
  <label
    className="label"
    htmlFor="This is a name"
  >
    This is a label
  </label>
  <div
    className="checkboxWrapper"
  >
    <input
      checked={false}
      className="checkbox"
      id="This is a name"
      name="This is a name"
      type="checkbox"
    />
    <label
      className="checkboxLabel"
      htmlFor="This is a name"
      style={
        Object {
          "background": undefined,
        }
      }
    >
      <span
        className="checkboxButton"
      />
    </label>
  </div>
</div>
`;

exports[`the InputText component renders the same as last time 1`] = `
<div
  className="wrapper"
>
  <label
    className="label"
    htmlFor="This is a name"
  >
    This is a label
  </label>
  <input
    className="text"
    id="This is a name"
    name="This is a name"
    type="text"
    value="This is a value"
  />
</div>
`;
